wire.md

```diff
@@ -0,0 +1,1008 @@
+
+A value field MUST NOT exceed 4096 bytes (4 kibibytes) in length.
+
+The valid bytes for a value depends on the key. See the table below.
+
+The following keys SHOULD be supported:
+
+key       | value format | desc
+----------|--------------|---------------------------------------
+`name`    | UTF-8        | the name this user wishes to be known as. Default value is "" (empty string).
+
+To save space, a client MAY discard older versions of these messages for a
+particular user.
+
+#### 6.2.5 `post/topic`
+
+Set a topic for a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+`topic_len`    | `varint`           | length of the topic field, in bytes
+`topic`        | `u8[topic_len] `   | topic content (UTF-8)
+
+`post_type` MUST be set to `3`.
+
+A `topic` field MUST be a valid UTF-8 string, between 0 and 512 codepoints. A
+topic of length zero MUST be considered as the current topic being cleared to
+the empty string, "".
+
+#### 6.2.6 `post/join`
+
+Publicly announce membership in a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+
+`post_type` MUST be set to `4`.
+
+#### 6.2.7 `post/leave`
+
+Publicly announce termination of membership in a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+
+`post_type` MUST be set to `5`.
+
+### 6.3 Message Formats
+
+#### 6.3.1 Message Header
+
+All messages MUST begin with the following header fields:
+
+field         | type     | desc
+--------------|----------|-------------------------------------------------------------
+`msg_len`     | `varint` | number of bytes in rest of message, not including the `msg_len` field
+`msg_type`    | `varint` | a type identifier for the message, which controls which fields follow this header
+`circuit_id`  | `u8[4]`  | id of a circuit for an established path, or `[0,0,0,0]` for no circuit
+`req_id`      | `u8[4]`  | unique id of this request (random)
+
+Message-specific fields follow after the `req_id`.
+
+Each request and response type has a unique `msg_type` (see below), which
+controls which fields will immediately follow this header.
+
+Clients encountering a `msg_type` they do not know how to parse MUST ignore and
+discard it.
+
+The `circuit_id` field is not currently in use, and MUST be set to all zeros.
```

Comment:
Then perhaps the desc in the table should simply say that it always is `[0, 0, 0, 0]` instead of introducing the concept of a circuit id if it never gets defined and/or used (which I'm assuming it won't at this point).

Reply:
Agreed.